### PR TITLE
Fix Line2 Raycasting bounds margin computation to support Orthographic Camera 

### PR DESCRIPTION
**Pull request by gkjohnson**
Related issue: #22782

**Description**

This PR adjusts the logic for computing the margin for the sphere and box bounds around Line2 to correctly (I believe) determine the world-space width of the thickness of the line at the closest point on the bounding sphere / box.

The logic for computing the margin is now encapsulated in `getWorldSpaceHalfWidth`. It now transforms a point at the given distance into clip space, adjusts the point by the `pixel width / resolution`, then transforms back into world space and returns the max of the xy values as the margin. Tested by copying the jsfiddle of the above issue into one of the fat_lines three.js example pages for local dev.

@nkallen can you test this?

cc @WestLangley

### CODE REVIEW

**Comment by mrdoob** [comment]
@WestLangley Looks good?

**Comment by nkallen** [comment]
I checked out this branch in my own codebase and the new behavior is good.

File: examples/jsm/lines/LineSegments2.js [hunk 1]
```diff
@@ -138,18 +146,8 @@ class LineSegments2 extends Mesh {
...
-		// transform back into world space to get world offset. Note clip space is [-1, 1] so full
-		// width does not need to be halved.
-		_clipToWorldVector.set( 0, 0, - distanceToBox, 1.0 ).applyMatrix4( camera.projectionMatrix );
-		_clipToWorldVector.multiplyScalar( 1.0 / _clipToWorldVector.w );
-		_clipToWorldVector.x = lineWidth / resolution.width;
-		_clipToWorldVector.y = lineWidth / resolution.height;
-		_clipToWorldVector.applyMatrix4( camera.projectionMatrixInverse );
-		_clipToWorldVector.multiplyScalar( 1.0 / _clipToWorldVector.w );
-
 		// increase the sphere bounds by the worst case line screen space width
```
Comment (nkallen) [inline examples/jsm/lines/LineSegments2.js:149]: "sphere" in the comment is incorrect

Comment (gkjohnson) [reply examples/jsm/lines/LineSegments2.js]: Fixed, thanks!

**Comment by WestLangley** [comment]
>@WestLangley Looks good?

I trust @gkjohnson on this feature. :-)

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**